docs/NOW-spec.md

```diff
@@ -538,134 +271,79 @@ The NOW_VARBUF structure is used to represent variable-length buffers.
-
-##### NOW_LRGBUF
-
-The NOW_LRGBUF structure is used to represent variable-length buffers.
-
-<table class="byte-layout">
-    <thead>
         <tr>
-            <th>0</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th><th>6</th><th>7</th>
-            <th>8</th><th>9</th><th>10</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th>
-            <th>6</th><th>7</th><th>8</th><th>9</th><th>20</th><th>1</th><th>2</th><th>3</th>
-            <th>4</th><th>5</th><th>6</th><th>7</th><th>8</th><th>9</th><th>30</th><th>1</th>
+            <td colspan="8">msgClass</td>
+            <td colspan="8">msgType</td>
+            <td colspan="16">msgFlags</td>
         </tr>
-    </thead>
-    <tbody>
         <tr>
-            <td colspan="32">size</td>
+            <td colspan="16">versionMajor</td>
+            <td colspan="16">versionMinor</td>
         </tr>
         <tr>
-            <td colspan="32">data (variable)</td>
+            <td colspan="32">systemCapset</td>
         </tr>
-    </tbody>
-</table>
-
-**size (4 bytes)**: A 32-bit unsigned integer containing the buffer size.
-
-**data (variable)**: The buffer data, whose size is given by the size field.
-
-#### NOW_HEADER
-
-The NOW_HEADER structure is the header common to all NOW protocol messages.
-
-<table class="byte-layout">
-    <thead>
         <tr>
-            <th>0</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th><th>6</th><th>7</th>
-            <th>8</th><th>9</th><th>10</th><th>1</th><th>2</th><th>3</th><th>4</th><th>5</th>
-            <th>6</th><th>7</th><th>8</th><th>9</th><th>20</th><th>1</th><th>2</th><th>3</th>
-            <th>4</th><th>5</th><th>6</th><th>7</th><th>8</th><th>9</th><th>30</th><th>1</th>
+            <td colspan="32">sessionCapset</td>
         </tr>
-    </thead>
-    <tbody>
         <tr>
-            <td colspan="32">msgSize</td>
+            <td colspan="32">execCapset</td>
         </tr>
         <tr>
-            <td colspan="8">msgClass</td>
-            <td colspan="8">msgType</td>
-            <td colspan="16">msgFlags</td>
+            <td colspan="32">heartbeatInterval(optional)</td>
         </tr>
     </tbody>
 </table>
 
 **msgSize (4 bytes)**: The message size, excluding the header size (8 bytes).
 
-**msgClass (1 byte)**: The message class.
+**msgClass (1 byte)**: The message class (NOW_NEGOTIATION_MSG_CLASS_ID).
 
-| Flag                            | Meaning              |
-|---------------------------------|----------------------|
-| NOW_SYSTEM_MSG_CLASS_ID<br>0x11 | System message class |
-| NOW_SESSION_MSG_CLASS_ID<br>0x12 | Session message class |
-| NOW_EXEC_MSG_CLASS_ID<br>0x13 | Exec message class |
+**msgType (1 byte)**: The message type (NOW_NEGOTIATION_CAPSET_MSG_ID).
 
-**msgType (1 byte)**: The message type, specific to the message class.
+**msgFlags (2 bytes)**: Message flags.
 
-**msgFlags (2 bytes)**: The message flags, specific to the message type and class.
+| Flag | Meaning |
+|-------|---------|
```

Comment:
Capabilities were moved to a separate message class. Main changes:
- Added capsets for session & system commands
- Added major & minor protocol version fields
- Added possibility to specify session heartbeat timeout (new feature)